modules/stochastic_tools/src/actions/AddSurrogateAction.C

```diff
@@ -27,5 +29,12 @@ AddSurrogateAction::AddSurrogateAction(InputParameters params) : MooseObjectActi
 void
 AddSurrogateAction::act()
 {
-  _problem->addUserObject(_type, _name, _moose_object_pars);
+  if (_current_task == "add_trainer")
+    _problem->addUserObject(_type, _name, _moose_object_pars);
+  else if (_current_task == "add_surrogate")
+  {
+    std::shared_ptr<SurrogateModel> model =
+        _factory.create<SurrogateModel>(_type, _name, _moose_object_pars);
+    _problem->theWarehouse().add(model);
+  }
```

Comment:
I just want to know your response to this comment before I approve.

Reply:
I gave this some thought and even tried to implement a check that would examine the type of object that created the data, but it was a bit too heavy handed. 

The main reason is that limiting the loaded data to some sort of name, such as the type, eliminates the ability for users to customize a trainer without registering the new name to the model in some way.

The current implementation just requires the the correct data structures and names exist. It will produce an error if the names are not found, although this could be improved (we an make an issue for this). This also allows a trainer to be customized, replaced completely, or inform multiple models.

Reply:
Ah, I think I see now. The data being saved only contains information on specific "elements", it doesn't have a descriptor for what the over arching type is. I like this for now, makes things modular. I guess if we get complaints from high-level users that they don't understand the error messages coming from loading the data, we can readdress this.

Reply:
I will add a note in the project page, we can address this a bit more when/if we have the funding/time to do it.